### DISCUSSION TITLE
Should we introduce a setting to determine if the port is a library or application

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#143] [Q&A]
Many of our libraries currently append a set of extra FLAGS to the .env  like in the case of openssl: https://github.com/ZOSOpenTools/opensslport/blob/main/buildenv#L35
```
if [ ! -z "\$ZOPEN_IN_ZOPEN_BUILD" ]; then
  export ZOPEN_EXTRA_CFLAGS="\${ZOPEN_EXTRA_CFLAGS} -I\$PWD/include"
  export ZOPEN_EXTRA_CXXFLAGS="\${ZOPEN_EXTRA_CXXFLAGS} -I\$PWD/include"
  export ZOPEN_EXTRA_LDFLAGS="\${ZOPEN_EXTRA_LDFLAGS} -L\$PWD/lib"
  export ZOPEN_EXTRA_LIBS="\${ZOPEN_EXTRA_LIBS} -lssl -lcrypto"

  #
  # These are required for wget
  #
  export OPENSSL_CFLAGS="-I\$PWD/include"
  export OPENSSL_LDFLAGS="-L\$PWD/lib"
  export OPENSSL_LIBS="-lssl -lcrypto"
fi
```

We could simplify this via `zopen generate` by asking what type of project it is (library, application, or both).

This would then set a variable in the buildenv as follows:
`ZOPEN_PORT_TYPE="LIBRARY"` or `ZOPEN_PORT_TYPE="APPLICATION" ` `ZOPEN_PORT_TYPE="APPLICATION_AND_LIBRARY"`

We would also introduce another setting for the installed library names
`ZOPEN_PORT_LIB_NAMES="ssl crypto"`

If LIBRARY is specified, then we can automatically add the ZOPEN_EXTRA_CFLAGS, CXXFLAGS, LDFLAGS, and LIBS to the .env.

We could also use this setting to help categorize the ports in https://github.com/ZOSOpenTools/meta/blob/main/docs/Latest.md

**Comment by MikeFultonDev** [comment]
Could we perhaps have an env var that says the name of the library? Then if you don’t specify it, it is not a library but if you do, it is? Then you can use that to generate the linker options.

**Reply by MikeFultonDev** [reply]
@IgorTodorovskiIBM sounds like overall thumbs up on this - could we have the env var be called ZOPEN_LIBRARY_NAMES and if it's specified it should be the actual libraries it provides, e.g. ZOPEN_LIBRARY_NAMES='ssl crypto' ? Then it's just one env var instead of 2?

**Comment by MikeFultonDev** [comment]
We have now sorted this out I believe. @IgorTodorovskiIBM re-open if you disagree (and perhaps create an issue now in meta for it)
